modules/sample/src/test/scala/core/Http4s/Http4sRoundTripTest.scala

```diff
@@ -0,0 +1,269 @@
+      def findPetsByStatusEnum(status: sdefs.definitions.PetStatus) = ???
+      def findPetsByTags(tags: Iterable[String])        = ???
+      def getPetById(petId: Long)                       = ???
+      def updatePet(body: sdefs.definitions.Pet)                    = ???
+      def updatePetWithForm(petId: Long, name: Option[String] = None, status: Option[String] = None) = ???
+      def uploadFile(
+          petId: PositiveLong,
+          additionalMetadata: Option[String] = None,
+          file: Option[fs2.Stream[IO, Byte]] = None,
+          file2: fs2.Stream[IO, Byte],
+          file3: fs2.Stream[IO, Byte],
+          longValue: Long,
+          customValue: PositiveLong,
+          customOptionalValue: Option[PositiveLong] = None) = ???
+    })
+
+    val petClient = PetClient.httpClient(Client.fromHttpService(httpService))
+
+    val result = petClient.deletePet(petId, Some(true), Some(cdefs.definitions.PetStatus.Pending), Some(apiKey)).attempt.unsafeRunSync()
+    result
+      .left
+      .foreach({ err =>
+        throw new TestFailedException(err.toString, 11)
+      })
+  }
+
+  test("round-trip: File uploads") {
+    val petId: Long    = 123L
+    val apiKey: String = "foobar"
+    val httpService = new PetResource().routes(new PetHandler[IO] {
+      def addPet(body: sdefs.definitions.Pet) = ???
+      def deletePet(_petId: Long,
+                                                                 includeChildren: Option[Boolean],
+                                                                 status: Option[sdefs.definitions.PetStatus],
+                                                                 _apiKey: Option[String] = None)                                                          = ???
+      def findPetsByStatus(status: Iterable[String])                                                  = ???
+      def findPetsByStatusEnum(status: sdefs.definitions.PetStatus)                                           = ???
+      def findPetsByTags(tags: Iterable[String])                                                        = ???
+      def getPetById(petId: Long)                                                                           = ???
+      def updatePet(body: sdefs.definitions.Pet)                                                                         = ???
+      def updatePetWithForm(petId: Long, name: Option[String] = None, status: Option[String] = None) = ???
+      def uploadFile(
+          petId: PositiveLong,
+          additionalMetadata: Option[String] = None,
+          file: Option[fs2.Stream[IO, Byte]] = None,
+          file2: fs2.Stream[IO, Byte],
+          file3: fs2.Stream[IO, Byte],
+          longValue: Long,
+          customValue: PositiveLong,
+          customOptionalValue: Option[PositiveLong] = None) = {
+        //val f1Length = file.flatMap({ case (f, _, _) => if (f.exists) { Some(f.length) } else None })
+        //val f2Length = if (file2._1.exists) { Some(file2._1.length) } else None
```

Comment:
Is there something useful that can be done to consume the provided streams?

Reply:
Added consumption of the provided streams.